### PR TITLE
updated contributing.md file

### LINKED ISSUE #674: Update contribution.md File
## Description

The current contribution guidelines in the contribution.md file need to be updated to better reflect our project's development process and standards. This will ensure that contributors have clear and up-to-date information on how to participate in our project effectively.

### PR DESCRIPTION
**Pull request by andoriyaprashant**
## Description 

This pull request updates the contributing.md file to provide clearer and more detailed guidelines for contributors to the project. The changes made in this PR include:

- Improved the welcome message to make it more inviting to potential contributors.
- Clarified the steps for getting started with contributions.
- Added guidelines for coding style and best practices specific to Flutter.
- Encouraged meaningful commit messages and thorough testing.
- Mentioned the presence of a Code of Conduct in the project.

These updates aim to enhance the contribution process and make it more accessible to newcomers while ensuring code quality and consistency within the project.

### CODE REVIEW

File: .github/workflows/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,10 +1,73 @@
...
+- When your work is ready and complies with the project conventions, upload your changes to your fork:
+
+```
+# To push your work to your remote repository
+$ git push -u origin Branch_Name
+```
+
+## Step 7 : Pull Request
+- Go to your repository in browser and click on compare and pull requests. 
+- Add a title and description to your pull request that explains your contribution.
```
Comment (guyluz11) [inline .github/workflows/CONTRIBUTING.md:60]: Can be removed.
Just write in a nice way 
1. If working on an issue please notify by commenting on it, no need to get assigned.
2. Fork and open pr from dev branch 
3. If there is an issue associated please link it in the description of the pr

File: .github/workflows/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,10 +1,73 @@
...
+## Need more help?🤔
+You can refer to the following articles on basics of Git and Github and also contact the Project Mentors, in case you are stuck:
+- [Forking a Repo](https://help.github.com/en/github/getting-started-with-github/fork-a-repo)
+- [Cloning a Repo](https://docs.github.com/en/repositories/creating-and-managing-repositories/cloning-a-repository)
+- [How to create a Pull Request](https://opensource.com/article/19/7/create-pull-request-github)
+- [Getting started with Git and GitHub](https://towardsdatascience.com/getting-started-with-git-and-github-6fcd0f2d4ac6)
+- [Learn GitHub from Scratch](https://lab.github.com/githubtraining/introduction-to-github)
+
+## Tip from us😇
+Learning and understanding takes time, so don't worry if things seem challenging at first. So, do not worry at all. We know **you have got this**!💪
```
Comment (guyluz11) [inline .github/workflows/CONTRIBUTING.md:73]: Not needed.
You can refer to the wiki for more info 
https://github.com/CyBear-Jinni/cbj_app/wiki
Take a look and decide if it is the right place to add it

**Review by guyluz11 — APPROVED** [review]

File: .github/workflows/CONTRIBUTING.md (hunk 2 above)
Comment (andoriyaprashant) [reply .github/workflows/CONTRIBUTING.md]: @guyluz11  As you mentioned, I have made the required changes, and you can check them
Thank you

File: .github/workflows/CONTRIBUTING.md [hunk 3]
```diff
@@ -1,10 +1,27 @@
...
+# Submitting Contributions👩‍💻👨‍💻
+Below you will find the process and workflow used to review and merge your changes.
 
-These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to
-this document in a pull request.
+# Working on an Issue
+If you're interested in working on an issue, simply let us know by commenting on it. No need to wait for an assignment. We welcome your contributions!
 
-## First rule, please open pull request to dev branch and not to master
+# Fork and Pull Request
```
Comment (guyluz11) [inline .github/workflows/CONTRIBUTING.md:11]: Subtitles can be smaller than the main title

File: .github/workflows/CONTRIBUTING.md [hunk 4]
```diff
@@ -1,10 +1,27 @@
...
+Please fork the repository and open your pull request from the `dev` branch. This helps us maintain a clean and organized contribution process.
+
+# Link to Associated Issues
+If your pull request addresses a specific issue, make sure to include a link to that issue in the pull request description. This helps us keep track of the context and purpose of your contribution.
+# Pull Request
+- Go to your repository in browser and click on compare and pull requests. 
+- Add a title and description to your pull request that explains your contribution.
+
+- 🥳:tada: Your Pull Request has been submitted and will be reviewed by the moderators and merged.
```
Comment (guyluz11) [inline .github/workflows/CONTRIBUTING.md:20]: Can be deleted

File: .github/workflows/CONTRIBUTING.md [hunk 5]
```diff
@@ -1,10 +1,27 @@
...
+
+## Need more help?🤔
+
+For additional information and resources, you can refer to our [Wiki](https://github.com/CyBear-Jinni/cbj_app/wiki). It contains valuable insights and documentation that can aid you in your contributions.
```
Comment (guyluz11) [inline .github/workflows/CONTRIBUTING.md:24]: Let's refer here also the discord server.

You can take from the readme file the link + the logo

**Review by guyluz11 — CHANGES_REQUESTED** [review]

File: .github/workflows/CONTRIBUTING.md (hunk 5 above)
Comment (andoriyaprashant) [reply .github/workflows/CONTRIBUTING.md]: Done

**Review by guyluz11 — APPROVED** [review]

**Comment by guyluz11** [comment]
Thanks that looks great 😃

**Pull request merged**